frontend/javascripts/admin/admin_rest_api.ts

```diff
@@ -2072,10 +2072,31 @@ export function getExistingExperienceDomains(): Promise<ExperienceDomainList> {
 }
 
 export async function isInMaintenance(): Promise<boolean> {
-  const info = await Request.receiveJSON("/api/maintenance", {
-    doNotInvestigate: true,
-  });
-  return info.isMaintenance;
+  const allMaintenances: Array<MaintenanceInfo> = await Request.receiveJSON(
+    "/api/maintenances/listCurrentAndUpcoming",
+  );
+  const currentEpoch = Date.now();
+  const currentMaintenance = allMaintenances?.find(
+    (maintenance) => maintenance.startTime < currentEpoch && maintenance.endTime > currentEpoch,
+  );
+  return currentMaintenance != null;
+}
+
+export type MaintenanceInfo = {
+  startTime: number;
+  endTime: number;
+  id: string;
+  message: string;
+};
+
+export async function listCurrentAndUpcomingMaintenances(): Promise<Array<MaintenanceInfo>> {
+  const allMaintenances: Array<MaintenanceInfo> = await Request.receiveJSON(
+    "/api/maintenances/listCurrentAndUpcoming",
+  );
+  const currentEpoch = Date.now();
+  return allMaintenances
+    ?.filter((maintenance) => maintenance.endTime > currentEpoch)
```

Comment:
1) see my other comment regarding the `?`
2) why is this filter necessary? shouldn't the backend only return current and upcoming entries (according to the name), anyway?

Reply:
oh no, that slipped through!